[CI] XPackRestIT test {p0=transform/transforms_crud/Test transform where source query is invalid} failing

**Build Scans:**
- [elasticsearch-pull-request #83394 / rest-compatibility](https://gradle-enterprise.elastic.co/s/njic6ojs6i7gk)
- [elasticsearch-intake #26229 / rest-compat](https://gradle-enterprise.elastic.co/s/stnbmb6aqcgdk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestCompatTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_crud/Test transform where source query is invalid}" -Dtests.seed=DCE12F60071F4890 -Dtests.locale=kam -Dtests.timezone=Pacific/Funafuti -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Dtransform%2Ftransforms_crud%2FTest%20transform%20where%20source%20query%20is%20invalid%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [transform/transforms_crud:584]: the error message was expected to match the provided regex but didn't
Expected: No handler for type \[bad-type\] declared on runtime field \[rt-field\]
     but: was "{root_cause=[{type=validation_exception, reason=Validation Failed: 1: Failed to test query, received status: BAD_REQUEST;, stack_trace=org.elasticsearch.ElasticsearchException$1: Validation Failed: 1: Failed to test query, received status: BAD_REQUEST;\n\tat org.elasticsearch.server@9.2.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:798)\n\tat org.elasticsearch.server@9.2.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:712)\n\tat org.elasticsearch.server@9.2.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:202)\n\tat org.elasticsearch.server@9.2.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:161)\n\tat o
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {p0=transform/transforms_crud/Test transform where source query is invalid}
- [main] 2 failures in test test {p0=transform/transforms_crud/Test transform where source query is invalid} (0.6% fail rate in 328 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -509,9 +509,6 @@ tests:
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   method: test {yaml=update/100_synthetic_source/keyword}
   issue: https://github.com/elastic/elasticsearch/issues/132110
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=transform/transforms_crud/Test transform where source query is invalid}
-  issue: https://github.com/elastic/elasticsearch/issues/132111
 - class: org.elasticsearch.index.engine.MergeWithLowDiskSpaceIT
   method: testRelocationWhileForceMerging
   issue: https://github.com/elastic/elasticsearch/issues/131789

